sources/vcf_data.py

```diff
@@ -0,0 +1,165 @@
+r"""
+uBiome fastq data extraction.
+
+Copyright (C) 2016 PersonalGenomes.org
+
+This software is shared under the "MIT License" license (aka "Expat License"),
+see LICENSE.TXT for full license text.
+
+May be used on the command line. For example, the following command:
+
+    python -m sources.ubiome ~/Downloads/uBiome.zip files
+
+Will assemble processed data sets in files/
+"""
+import bz2
+import json
+import os
+import tempfile
+import vcf
+
+import requests
+
+from boto.s3.connection import S3Connection
+
+from data_retrieval.files import get_remote_file, mv_tempfile_to_output
+
+
+def s3_connection():
+    """
+    Get an S3 connection using environment variables.
+    """
+    key = os.getenv('AWS_ACCESS_KEY_ID')
+    secret = os.getenv('AWS_SECRET_ACCESS_KEY')
+
+    if not (key and secret):
+        raise Exception('You must specify AWS credentials.')
+
+    return S3Connection(key, secret)
+
+
+def verify_vcf(input_filepath, sentry=None, username=None):
+    """
+    Verify that this is a VCF file.
+    """
+    if input_filepath.endswith('.vcf.gz'):
+        input_vcf = vcf.Reader(filename=input_filepath, compressed=True)
+    elif input_filepath.endswith('.vcf.bz2'):
+        vcf_file = bz2.BZ2File(input_filepath)
+        input_vcf = vcf.Reader(vcf_file)
+    elif input_filepath.endswith('.vcf'):
+        input_vcf = vcf.Reader(filename=input_filepath)
+    else:
+        if sentry:
+            sentry_msg = ('vcf_data: input file does not appear to be VCF. '
+                          'Username: {}'.format(username))
+            sentry.captureMessage(sentry_msg)
+        raise ValueError('Input file does not appear to be VCF.')
+    # Check that it can advance one record without error.
+    input_vcf.next()
```

Comment:
if this raises an exception do we propagate the error back or does it just blow up?